R/inven.R

```diff
@@ -67,3 +67,218 @@ inventory <- function(x,obj,..., .strict = FALSE) {
   
   return(invisible(x))
 }
+
+#' Check input data set names against model parameters
+#' 
+#' Use this function to check names of input data sets against parameters that 
+#' have been assigned different tags. Assignment is made in the model 
+#' specification file. This is useful to alert the user to misspelled or 
+#' otherwise misspecified parameter names in input data sets. See [param_tags()]
+#' for information on associating tags with parameters.
+#' 
+#' @param data a data frame or other object with names to check.
+#' @param x a model object.
+#' @param check_covariates logical; if `TRUE`, check `data` for parameter names 
+#' carrying the `covariates` tag.
+#' @param check_inputs logical; if `TRUE`, check `data` for parameter names 
+#' carrying the `input` tag. 
+#' @param tags a character vector of user-defined parameter tags to require 
+#' in `data`; this may be a comma- or space-separated string (e.g. 
+#' `"tag1,tag2"`).
+#' @param strict if `TRUE`, generate an error when `data` is missing some
+#' expected column names; otherwise, issue a warning.
+#' @param inform if `TRUE`, inform the user when `data` is missing some 
```

Comment:
As a user, I tend to find dependent boolean flags like `strict` and `inform` more confusing than having one option with multiple modes: `if_missing = c("error", "warn", "inform")`.  Not sure how users feel the same though.

Reply:
I understand; I wondered about that approach for `check_covariates` and `check_input` too, but wanted to force the user to opt out of both if they really wanted. 

`inform` was a late addition and I just put it in as another boolean. I can pull those into a single argument.

Reply:
This has been re-factored; good call.